riscv/src/compiler.rs

```diff
@@ -386,6 +387,20 @@ fn store_data_objects<'a>(
                             ]);
                         }
                     }
+                    DataValue::Offset(a, b) => {
+                        object_code.push(format!("addr <=X= 0x{pos:x};"));
+                        assert!(!positions.contains_key(a));
+                        assert!(!positions.contains_key(b));
```

Comment:
so in the test that I had, this wasn't reached because the asm code that uses this was removed by the unreachable filter. If I leave it in this breaks. Fixing it now